### PR TITLE
Add alias "allowlist_externals" to "whitelist_externals"

### LINKED ISSUE #1491: Editorial: consider offering alternative to whitelist_externals
Using whitelist (and its counterpart) isn't exactly welcoming. While migrating away from it entirely might be hard due to compatibility considerations, it'd be nice if configuration files from projects that build on this technology could avoid it.

See https://tools.ietf.org/html/draft-knodel-terminology for additional context.

### PR DESCRIPTION
**Pull request by dajose**
closes #1491

### CODE REVIEW

**Comment by dajose** [comment]
I think the macOS checks failure is not related with the PR, can someone give me a hand there?

File: src/tox/config/__init__.py [hunk 1]
```diff
@@ -747,8 +747,13 @@ def passenv(testenv_config, value):
         "eventual passenv setting.",
     )
 
+    # FIXME remove whitelist in favor of allowlist_external
     parser.add_testenv_attribute(
-        name="whitelist_externals",
+        name="whitelist_externals", type="line-list", help="DEPRECATED: use allowlist_external",
```
Comment (pawamoy) [inline src/tox/config/__init__.py:752]: Missing `s` at the end.

File: src/tox/config/__init__.py [hunk 2]
```diff
@@ -747,8 +747,13 @@ def passenv(testenv_config, value):
         "eventual passenv setting.",
     )
 
+    # FIXME remove whitelist in favor of allowlist_external
```
Comment (pawamoy) [inline src/tox/config/__init__.py:750]: (Missing `s` at the end)

File: docs/changelog/1491.deprecation.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Whitelist (as used on whitelist_externals) has a racial weight, allowlist_externals should be preferred. - by :user:`dajose`
```
Comment (gaborbernat) [inline docs/changelog/1491.deprecation.rst:1]: This needs to expand on the fact that whitelist_externals is still supported, however now the allowlist_externals alias has been added.

File: docs/config.rst [hunk 4]
```diff
@@ -325,15 +325,15 @@ Complete list of settings that you can put into ``testenv*`` sections:
...
     Each line specifies a command name (in glob-style pattern format)
     which can be used in the ``commands`` section without triggering
     a "not installed in virtualenv" warning.  Example: if you use the
-    unix ``make`` for running tests you can list ``whitelist_externals=make``
-    or ``whitelist_externals=/usr/bin/make`` if you want more precision.
+    unix ``make`` for running tests you can list ``allowlist_externals=make``
+    or ``allowlist_externals=/usr/bin/make`` if you want more precision.
     If you don't want tox to issue a warning in any case, just use
-    ``whitelist_externals=*`` which will match all commands (not recommended).
+    ``allowlist_externals=*`` which will match all commands (not recommended).
```
Comment (gaborbernat) [inline docs/config.rst:336]: This needs to still mention that whitelist_external is still is accepted.

File: src/tox/config/__init__.py (hunk 2 above)
Comment (gaborbernat) [reply src/tox/config/__init__.py:750]: We don't use fixmes in this project but instead prefer the issue tracker, please remove this. We could not remove it either way before the next major version as that would be breaking change.

File: tests/unit/test_venv.py [hunk 5]
```diff
@@ -433,6 +433,10 @@ def test_install_command_not_installed(newmocksession):
 
 
 def test_install_command_whitelisted(newmocksession):
+    """
+    Deprecated: Remove in favor of test_install_command_allowlisted when
```
Comment (gaborbernat) [inline tests/unit/test_venv.py:437]: Remove this docstring, the migration complete will not happen before the rewrite.

File: src/tox/venv.py [hunk 6]
```diff
@@ -223,7 +223,7 @@ def is_allowed_external(self, p):
         if tox.INFO.IS_WIN:
             tryadd += [os.path.normcase(x) for x in os.environ["PATHEXT"].split(os.pathsep)]
             p = py.path.local(os.path.normcase(str(p)))
-        for x in self.envconfig.whitelist_externals:
+        for x in self.envconfig.allowlist_externals + self.envconfig.whitelist_externals:
```
Comment (gaborbernat) [inline src/tox/venv.py:226]: I'm not happy to allow specifying both of them. We should instead allow only one, and raise if both set. And then pick only the one that is set.

**Comment by dajose** [comment]
@gaborbernat can you give me a hand on why it is failing? 

I think it is not related with the changes: https://dev.azure.com/toxdev/tox/_build/results?buildId=2498&view=logs&j=6f785b48-ed19-5349-e6cb-bb6d29a19c8b&t=4621fbdb-f74d-5491-ece0-e6cd5b96369c&l=704

**Comment by asottile** [comment]
re-triggered CI, seems to be doing better now

**Pull request merged**